scipy/stats/_multivariate.py

```diff
@@ -5066,3 +5067,185 @@ def rvs(self, size=1, random_state=None):
         method.__doc__, mhg_docdict_noparams)
     method.__doc__ = doccer.docformat(method.__doc__,
                                       mhg_docdict_params)
+
+
+class random_direction_gen(multi_rv_generic):
+    r"""A vector-valued random direction.
+
+    Return a random direction (unit vector).
+
+    The `dim` keyword specifies the dimension N.
+
+    Methods
+    -------
+    rvs(dim=None, size=1, random_state=None)
+        Draw random directions.
+
+    Parameters
+    ----------
+    dim : scalar
+        Dimension of directions.
+    seed : {None, int, np.random.RandomState, np.random.Generator}, optional
+        Used for drawing random variates.
+        If `seed` is `None`, the `~np.random.RandomState` singleton is used.
+        If `seed` is an int, a new ``RandomState`` instance is used, seeded
+        with seed.
+        If `seed` is already a ``RandomState`` or ``Generator`` instance,
+        then that object is used.
+        Default is `None`.
+
+    Notes
+    -----
+    This distribution generates unit vectors uniformly distributed on
+    the surface of a sphere. These can be interpreted as random directions.
+
+    For example, if `dim=3`, 3D vectors from the surface of math:`S^2`
+    will be sampled.
+
+    References
+    ----------
+    .. [1] Marsaglia, G. (1972). "Choosing a Point from the Surface of a
+           Sphere". Annals of Mathematical Statistics. 43 (2): 645-646.
```

Comment:
I don't usually comment much on `stats` PRs, but I did want to chime in here for a few reasons because I have cited that work before from the computational geometry side and:

1. If you look at `scipy/spatial/tests/test_spherical_voronoi.py` there is a `_sample_sphere` function--if you/the `stats` folks consider the work here more robust, or at least equivalent, perhaps we should open an issue to remind us to replace the *ad hoc* private implementation used for testing there with the "real" version being worked on here?
2. That said, from the `spatial` side, I do find this approach a bit confusing, perhaps because I'm not used to using the `stats` API as much. It looks like two calls are needed?

```python
random_direction_dist = random_direction(dim)
samples = random_direction_dist.rvs(size)
```

whereas for the isolated purpose of "give me some randomly distributed points on a sphere," I do find it much easier/more intuitive to do something like I did over in that test module with a single function:

```python
points = _sample_sphere(n, dim, seed=0)
```

I'm guessing the current design is mostly for consistency with other parts of `stats` and/or to enable flexibility for sampling beyond my crude use cases.

Reply:
1. The comments claim that the special cases are for efficiency. I'm not sure if that has been tested or if the efficiency gains are worthwhile. For sure, for the `dim>3` case, using `rng.standard_normal()` is better than using `rng.multivariate_normal()`.
2. No, just `samples = random_direction_dist.rvs(dim, size=size)` suffices.

Reply:
Good point. No idea why I directly went to `multivariate_normal`. Will simplify this. With `rng.standard_normal()`, only the 2D case is faster for large sample sizes.

Reply:
Ah, for 2 I was looking at `test_samples`, which seems to use two calls.